etc/kernel-launchers/python/scripts/launch_ipykernel.py

```diff
@@ -590,6 +605,8 @@ def start_ipython(
     if cluster_type == "spark" and spark_init_mode == "none":
         cluster_type = "none"
```

Comment:
Since we're now relying on `cluster_type` in the listener (sub)process, this needs to be moved to L571 (outside the block that launches the sub-process), otherwise, the subprocess will believe non-spark kernelspecs have access to the spark context and will result in attribute errors indicating that `cancelAllJobs` is not available...
![Screen Shot 2022-06-23 at 11 24 58 AM](https://user-images.githubusercontent.com/22599560/175370881-6b2c1578-c76a-42fc-bef7-cdb4f5c353b5.png)